base/sparse/sparsematrix.jl

```diff
@@ -193,6 +193,23 @@ end
 copy(S::SparseMatrixCSC) =
     SparseMatrixCSC(S.m, S.n, copy(S.colptr), copy(S.rowval), copy(S.nzval))
 
+function copy!{TvA, TiA, TvB, TiB}(A::SparseMatrixCSC{TvA,TiA},
+                                   B::SparseMatrixCSC{TvB,TiB})
+    # If the two matrices have the same number of elements then all the
+    # elements in A will be overwritten and we can simply copy the
+    # internal fields of B to A.
+    if prod(size(A)) == prod(size(B))
+        copy!(A.colptr, B.colptr)
```

Comment:
or given @mbauman's catch below, maybe not worth the trouble to do the specialized path when `size(A) != size(B)`